### ISSUE TITLE
Not handling escape on !: "Unknown binary operator" for "\!"

### ISSUE THREAD

**Issue by KevinMatte** [koalaman/shellcheck#1984]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2057.
- My shellcheck version (`shellcheck --version` or "online"): Included with PHPStorm
- [ X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [X ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

A backslash \ before a ! makes it clear the ! is NOT a special parameter to be expanded.
It shouldn't be flagged as a problem.

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

f=/tmp/myFile;

[ \! -f $f ]

```

#### Here's what shellcheck currently says:

Line 5:
[ \! -f $f ]
     ^-- SC2057: Unknown binary operator.

#### Here's what I wanted or expected to see:

No issues detected!

**Comment by AlanDeSmet** [comment]
Also happens with `"!"`, as in:

```
#! /bin/sh

if [ '!' -e outputfile ]; then 
        date > outputfile
fi
```
